packages/web3/src/address/address.test.ts

```diff
@@ -17,194 +17,316 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
-      '030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc'
-    )
+  describe('addressFromScript', () => {
+    it('should compute address from script', () => {
+      const script = randomBytes(20)
+      const address = addressFromScript(script)
+      expect(address).toBeTruthy()
+    })
   })
 
-  it('should compute address from public key', () => {
-    expect(publicKeyFromPrivateKey('91411e484289ec7e8b3058697f53f9b26fa7305158b4ef1a81adfbabcf090e45')).toBe(
-      '030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc'
-    )
-    expect(addressFromPublicKey('030f9f042a9410969f1886f85fa20f6e43176ae23fc5e64db15b3767c84c5db2dc')).toBe(
-      '1ACCkgFfmTif46T3qK12znuWjb5Bk9jXpqaeWt2DXx8oc'
-    )
+  describe('contractIdFromAddress', () => {
+    it('should convert between contract id and address', () => {
+      expect(addressFromContractId('1f6b937b935d7fac894fb22ffe2b974cae9c8c166501372f1b9155144e0ff4ae')).toBe(
+        'vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3'
+      )
+      expect(binToHex(contractIdFromAddress('vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3'))).toBe(
+        '1f6b937b935d7fac894fb22ffe2b974cae9c8c166501372f1b9155144e0ff4ae'
+      )
+    })
+
+    it('should throw error for non-contract addresses', () => {
+      expect(() => contractIdFromAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGFyu')).toThrow(
+        'Invalid contract address type: 0'
+      )
+    })
   })
 
-  it('should convert between contract id and address', () => {
-    expect(addressFromContractId('1f6b937b935d7fac894fb22ffe2b974cae9c8c166501372f1b9155144e0ff4ae')).toBe(
-      'vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3'
-    )
-    expect(binToHex(contractIdFromAddress('vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3'))).toBe(
-      '1f6b937b935d7fac894fb22ffe2b974cae9c8c166501372f1b9155144e0ff4ae'
-    )
+  describe('tokenIdFromAddress', () => {
+    it('should extract token id from addresses', () => {
+      expect(binToHex(tokenIdFromAddress('wCTeteGBeSEC54GpkS8jWBzYiYNTBUuTW3WzxGd9yExT'))).toBe(
+        '25469eb0d0d0a55deea832924547b7b166c70a3554fe321e81886d3c18f19d64'
+      )
+      expect(binToHex(tokenIdFromAddress('xrY8dxgVm38QCQXhiUFcivFutLFUNMoo8qu8vYf7wJps'))).toBe(
+        '3de370f893cb1383c828c0eb22c89aceb13fa56ddced1848db27ce7fa419c80c'
+      )
+    })
+
+    it('should throw error for non-contract addresses', () => {
+      expect(() => tokenIdFromAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toThrow(
+        'Invalid contract address type: 2'
+      )
+    })
+  })
+
+  describe('contractIdFromTx', () => {
+    it('should compute contract ID from txId and outputIndex', () => {
+      const txId = '0a38bc48fbb4300f1e305b201cd6129372d867122efb814d871d18c0bfe43b56'
+      const outputIndex = 0
+      const contractId = contractIdFromTx(txId, outputIndex)
+      expect(contractId).toBeTruthy()
```

Comment:
This is not verifying the contract id is correct, only that it returns something. We might need to create a contract id in integration test and verify it it matches what's returned by `contractIdFromTx(txId, outputIndex)`